### DISCUSSION TITLE
What's the correct way to implement window position in a window rule?

### DISCUSSION THREAD

**Discussion by guzz46** [labwc/labwc#1942] [Q&A]
I currently have the following for MPV

```
    <windowRules>
      <windowRule identifier="mpv" matchOnce="false">
        <skipTaskbar>no</skipTaskbar>
        <action name="MoveTo" x="981" y="564" />
        <action name="ToggleAlwaysOnTop"/>
      </windowRule>
```
Which works great, it opens MPV in the bottom right corner of the screen, except when I launch MPV in fullscreen, then it opens in fullscreen from that position, so most of the video will be off screen, if I open MPV and then enter fullscreen it works fine, the problem only happens when I launch MPV in fullscreen, is there another way of doing this or is this a bug of some sort? thanks

**Comment by johanmalm** [comment]
We should probably prevent `view_move_resize()` from doing anything with a view in fullscreen mode.

Might this help?

```
diff --git a/src/view.c b/src/view.c
index 1646dec..a43e560 100644
--- a/src/view.c
+++ b/src/view.c
@@ -493,6 +493,9 @@ void
 view_move_resize(struct view *view, struct wlr_box geo)
 {
 	assert(view);
+	if (view->fullscreen) {
+		return;
+	}
 	if (view->impl->configure) {
 		view->impl->configure(view, geo);
 	}
```

**Reply by guzz46** [reply]
I wouldn't know, I'm not a programmer unfortunately

**Reply by Consolatis** [reply]
> We should probably prevent `view_move_resize()` from doing anything with a view in fullscreen mode.
> 
> Might this help?

I think in general this is the right approach, AFAIR we only restrict movement of fullscreen windows when doing interactive move / resize currently.

I am not sure if this could interfere with changing resolutions, disabled outputs and so on though. Maybe we need that check in the actions or their direct calls (if available) instead. Also not sure how MoveToOutput, FitToOutput, MoveToCursor / AutoPlace and ResizeTo deal with full screen views.

**Comment by droc12345** [comment]
Johan, that might work. 
But it it doesn't then maybe look at the feasibility of setting x,y coords to 0,0 for fullscreen.
But I haven't looked at all the code, so this is just off the top of my head.
